src/utils.c

```diff
@@ -80,6 +80,54 @@ char *str_concat_many(int count, ...)
     return combined;
 }
 
+char *str_replace(char *search, char *replace, char *subject) {
```

Comment:
a few unit tests around this could be useful?

Reply:
Just added a unit test - see the latest commit.